### ISSUE TITLE
Flutter daemon stuck at starting on VSCode

### ISSUE THREAD

**Issue by namkazt** [flutter/flutter#143625]
### Steps to reproduce

1, Install flutter following guide in: https://docs.flutter.dev/get-started/install/windows/desktop?tab=vscode#install-the-flutter-sdk
2, Create new project
3, Check VSCode flutter extension

### Actual results

VS code daemon stuck at starting
![image](https://github.com/flutter/flutter/assets/6035916/5dcd66e2-9922-457f-9ddd-3966b2e73eac)

There is no devices in sidebar of flutter extension
![image](https://github.com/flutter/flutter/assets/6035916/72d04383-af69-4b15-91fd-88e7be3e8c98)

### Logs

<details open>
<summary>Logs</summary>

```console

!! ⚠️ PLEASE REVIEW THIS LOG FOR SENSITIVE INFORMATION BEFORE SHARING ⚠️ !!
Dart Code extension: 3.82.0
Flutter extension: 3.83.20240201 (not activated)

App: Visual Studio Code
App Host: desktop
Version: win 1.86.2


Logging Categories:
    FlutterDaemon

Sat Feb 17 2024 [10:12:47 GMT+0700 (Indochina Time)] Log file started
[10:12:47 AM] [FlutterDaemon] [Info] Spawning E:\works\flutter\bin\flutter.bat with args ["daemon"]
[10:12:47 AM] [FlutterDaemon] [Info] ..  in E:\works\flutter
[10:12:47 AM] [FlutterDaemon] [Info] ..  with {"toolEnv":{"FLUTTER_HOST":"VSCode","PUB_ENVIRONMENT":"vscode.dart-code","FLUTTER_ROOT":"E:\\works\\flutter"}}
[10:12:47 AM] [FlutterDaemon] [Info]     PID: 20352
[10:12:47 AM] [FlutterDaemon] [Info] <== [{"event":"daemon.connected","params":{"version":"0.6.1","pid":3244}}]
[10:12:47 AM] [FlutterDaemon] [Info] ==> [{"id":"1","method":"device.enable"}]
[10:12:47 AM] [FlutterDaemon] [Info] ==> [{"id":"2","method":"emulator.getEmulators"}]
[10:12:47 AM] [FlutterDaemon] [Info] ==> [{"id":"3","method":"daemon.getSupportedPlatforms","params":{"projectRoot":"E:\\works\\dp2\\rxapp\\frontend\\app"}}]
[10:12:47 AM] [FlutterDaemon] [Info] <== [{"event":"daemon.logMessage","params":{"level":"status","message":"Starting device daemon..."}}]
[10:15:29 AM] [FlutterDaemon] [Info] ==> [{"id":"4","method":"daemon.getSupportedPlatforms","params":{"projectRoot":"E:\\works\\dp2\\rxapp\\frontend\\app"}}]

```

</details>

### Flutter Doctor output

<details open>
<summary>Doctor output</summary>

```console

[√] Flutter (Channel stable, 3.19.0, on Microsoft Windows [Version 10.0.22621.3155], locale vi-VN)
    • Flutter version 3.19.0 on channel stable at E:\works\flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • FLUTTER_GIT_URL = https://github.com/flutter/flutter.git
    • Framework revision bae5e49bc2 (3 days ago), 2024-02-13 17:46:18 -0800
    • Engine revision 04817c99c9
    • Dart version 3.3.0
    • DevTools version 2.31.1

[√] Windows Version (Installed version of Windows is version 10 or higher)

[√] Android toolchain - develop for Android devices (Android SDK version 33.0.2)
    • Android SDK at C:\Users\namka\AppData\Local\Android\sdk
    • Platform android-33, build-tools 33.0.2
    • Java binary at: C:\Program Files\Android\Android Studio\jbr\bin\java
    • Java version OpenJDK Runtime Environment (build 17.0.7+0-b2043.56-10550314)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • CHROME_EXECUTABLE = C:\Program Files\BraveSoftware\Brave-Browser\Application\brave.exe

[√] Visual Studio - develop Windows apps (Visual Studio Build Tools 2019 16.11.32)
    • Visual Studio at C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools
    • Visual Studio Build Tools 2019 version 16.11.34301.259
    • Windows 10 SDK version 10.0.19041.0

[√] Android Studio (version 2023.1)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.7+0-b2043.56-10550314)

[√] VS Code (version 1.86.2)
    • VS Code at C:\Users\namka\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.83.20240201

[√] Connected device (3 available)
    • Windows (desktop) • windows • windows-x64    • Microsoft Windows [Version 10.0.22621.3155]
    • Chrome (web)      • chrome  • web-javascript • unknown
    • Edge (web)        • edge    • web-javascript • Microsoft Edge 121.0.2277.106

[√] Network resources
    • All expected network resources are available.

• No issues found!

```

</details>

**Comment by huycozy** [comment]
Hi @namkazt 
I see you are using `Flutter extension version 3.83.20240201` while the latest stable release is [v3.82.0](https://dartcode.org/releases/v3-82/). Are you using a pre-release version? Can you try switching to v3.82.0 to see if the issue persists?

If this issue is an issue with Flutter extension on VSCode, it should be reported to its [repository](https://github.com/Dart-Code/Dart-Code) for better support there.

**Comment by DanTup** [comment]
Could be the same issue we're debugging in https://github.com/Dart-Code/Dart-Code/issues/4840

For the user in there, it seems like the stdin stream is being closed so Flutter tools never gets the messages from the VS Code extension. It's not clear to me what's going on yet (and I can't reproduce it myself on Windows).

**Comment by namkazt** [comment]
> Could be the same issue we're debugging in [Dart-Code/Dart-Code#4840](https://github.com/Dart-Code/Dart-Code/issues/4840)
> 
> For the user in there, it seems like the stdin stream is being closed so Flutter tools never gets the messages from the VS Code extension. It's not clear to me what's going on yet (and I can't reproduce it myself on Windows).

thanks @DanTup, if there is anything i can do to help just tell me.

**Comment by namkazt** [comment]
anyway @DanTup  it suddenly showing device here:
![image](https://github.com/flutter/flutter/assets/6035916/3d37cf77-572f-4694-9317-f2caccf09073)

### Logs

<details open>
<summary>Logs</summary>

```console
!! ⚠️ PLEASE REVIEW THIS LOG FOR SENSITIVE INFORMATION BEFORE SHARING ⚠️ !!
Dart Code extension: 3.83.20240220
Flutter extension: 3.83.20240201 (not activated)

App: Visual Studio Code
App Host: desktop
Version: win 1.86.2


Logging Categories:
    FlutterDaemon

Wed Feb 21 2024 [00:55:17 GMT+0700 (Indochina Time)] Log file started
[12:55:17 AM] [FlutterDaemon] [Info] Spawning E:\works\flutter\bin\flutter.bat with args ["-v","daemon"]
[12:55:17 AM] [FlutterDaemon] [Info] ..  in E:\works\flutter
[12:55:17 AM] [FlutterDaemon] [Info] ..  with {"toolEnv":{"FLUTTER_HOST":"VSCode","PUB_ENVIRONMENT":"vscode.dart-code","ANDROID_HOME":"C:\\Users\\namka\\AppData\\Local\\Android","CHROME_EXECUTABLE":"C:\\Program Files\\BraveSoftware\\Brave-Browser\\Application\\brave.exe","FLUTTER_ROOT":"E:\\works\\flutter"}}
[12:55:17 AM] [FlutterDaemon] [Info]     PID: 21732
[12:55:18 AM] [FlutterDaemon] [Error] [ +302 ms] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
[   +9 ms] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterWebSdk' is not required, skipping update.
[        ] Artifact Instance of 'LegacyCanvasKitRemover' is not required, skipping update.
[   +5 ms] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[ +107 ms] Artifact Instance of 'MaterialFonts' is not required, skipping update.
[        ] Artifact Instance of 'GradleWrapper' is not required, skipping update.
[        ] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterWebSdk' is not required, skipping update.
[        ] Artifact Instance of 'LegacyCanvasKitRemover' is not required, skipping update.
[12:55:18 AM] [FlutterDaemon] [Error] [        ] Artifact Instance of 'FlutterSdk' is not required, skipping update.
[        ] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[  +17 ms] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FontSubsetArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'PubDependencies' is not required, skipping update.
[12:55:18 AM] [FlutterDaemon] [Info] <== [{"event":"daemon.connected","params":{"version":"0.6.1","pid":9320}}]
[12:55:18 AM] [FlutterDaemon] [Info] ==> [{"id":"1","method":"device.enable"}]
[12:55:18 AM] [FlutterDaemon] [Info] ==> [{"id":"2","method":"emulator.getEmulators"}]
[12:55:18 AM] [FlutterDaemon] [Info] ==> [{"id":"3","method":"daemon.getSupportedPlatforms","params":{"projectRoot":"E:\\works\\dp2\\rx"}}]
[12:55:18 AM] [FlutterDaemon] [Error] [ +173 ms] executing: C:\Program Files\Android\Android Studio\jbr\bin\java -version
[12:55:19 AM] [FlutterDaemon] [Error] [ +321 ms] Exit code 0 from: C:\Program Files\Android\Android Studio\jbr\bin\java -version
[12:55:19 AM] [FlutterDaemon] [Error] [        ] openjdk version "17.0.7" 2023-04-18
           OpenJDK Runtime Environment (build 17.0.7+0-b2043.56-10550314)
           OpenJDK 64-Bit Server VM (build 17.0.7+0-b2043.56-10550314, mixed mode)
[12:55:19 AM] [FlutterDaemon] [Info] <== [{"event":"daemon.logMessage","params":{"level":"status","message":"Starting device daemon..."}}]
[12:55:19 AM] [FlutterDaemon] [Info] <== [{"id":"1"}]
[12:55:19 AM] [FlutterDaemon] [Error] [  +48 ms] executing: C:\Users\namka\AppData\Local\Android\sdk\emulator\emulator.exe -list-avds
[12:55:19 AM] [FlutterDaemon] [Info] <== [{"id":"3","result":{"platforms":["web","android","ios","linux","macos","windows"],"platformTypes":{"web":{"isSupported":true},"android":{"isSupported":true},"ios":{"isSupported":true},"linux":{"isSupported":true},"macos":{"isSupported":true},"windows":{"isSupported":true},"fuchsia":{"isSupported":false,"reasons":[{"reasonText":"the Fuchsia feature is not enabled","fixText":"Run \"flutter config --enable-fuchsia\"","fixCode":"config"},{"reasonText":"the Fuchsia platform is not enabled for this project","fixText":"Run \"flutter create --platforms=fuchsia .\" in your application directory","fixCode":"create"}]},"custom":{"isSupported":false,"reasons":[{"reasonText":"the custom devices feature is not enabled","fixText":"Run \"flutter config --enable-custom-devices\"","fixCode":"config"}]},"windowsPreview":{"isSupported":false,"reasons":[{"reasonText":"the Preview Device feature is not enabled","fixText":"Run \"flutter config --enable-flutter-preview","fixCode":"config"}]}}}}]
[12:55:19 AM] [FlutterDaemon] [Error] [ +117 ms] executing: C:\Users\namka\AppData\Local\Android\sdk\platform-tools\adb.exe devices -l
[12:55:19 AM] [FlutterDaemon] [Info] <== [{"event":"device.added","params":{"id":"windows","name":"Windows","platform":"windows-x64","emulator":false,"category":"desktop","platformType":"windows","ephemeral":false,"emulatorId":null,"sdk":"Microsoft Windows [Version 10.0.22621.3155]","capabilities":{"hotReload":true,"hotRestart":true,"screenshot":false,"fastStart":false,"flutterExit":true,"hardwareRendering":true,"startPaused":true}}}]
[12:55:19 AM] [FlutterDaemon] [Info] <== [{"id":"2","result":[]}]
[12:55:19 AM] [FlutterDaemon] [Error] [ +196 ms] List of devices attached
[12:55:19 AM] [FlutterDaemon] [Info] <== [{"event":"device.added","params":{"id":"chrome","name":"Chrome","platform":"web-javascript","emulator":false,"category":"web","platformType":"web","ephemeral":false,"emulatorId":null,"sdk":"unknown","capabilities":{"hotReload":true,"hotRestart":true,"screenshot":false,"fastStart":false,"flutterExit":false,"hardwareRendering":true,"startPaused":true}}}]
[12:55:19 AM] [FlutterDaemon] [Info] <== [{"event":"device.added","params":{"id":"edge","name":"Edge","platform":"web-javascript","emulator":false,"category":"web","platformType":"web","ephemeral":false,"emulatorId":null,"sdk":"Microsoft Edge 121.0.2277.106","capabilities":{"hotReload":true,"hotRestart":true,"screenshot":false,"fastStart":false,"flutterExit":false,"hardwareRendering":true,"startPaused":true}}}]
[12:55:23 AM] [FlutterDaemon] [Error] [+4019 ms] executing: C:\Users\namka\AppData\Local\Android\sdk\platform-tools\adb.exe devices -l
[12:55:23 AM] [FlutterDaemon] [Error] [  +26 ms] List of devices attached
[12:55:27 AM] [FlutterDaemon] [Error] [+4015 ms] executing: C:\Users\namka\AppData\Local\Android\sdk\platform-tools\adb.exe devices -l
[12:55:27 AM] [FlutterDaemon] [Error] [  +27 ms] List of devices attached
[12:55:31 AM] [FlutterDaemon] [Error] [+4008 ms] executing: C:\Users\namka\AppData\Local\Android\sdk\platform-tools\adb.exe devices -l
[12:55:31 AM] [FlutterDaemon] [Error] [  +26 ms] List of devices attached
[12:55:35 AM] [FlutterDaemon] [Error] [+4011 ms] executing: C:\Users\namka\AppData\Local\Android\sdk\platform-tools\adb.exe devices -l
[12:55:35 AM] [FlutterDaemon] [Error] [  +26 ms] List of devices attached
[12:55:39 AM] [FlutterDaemon] [Error] [+4002 ms] executing: C:\Users\namka\AppData\Local\Android\sdk\platform-tools\adb.exe devices -l
[12:55:39 AM] [FlutterDaemon] [Error] [  +25 ms] List of devices attached
[12:55:43 AM] [FlutterDaemon] [Error] [+4002 ms] executing: C:\Users\namka\AppData\Local\Android\sdk\platform-tools\adb.exe devices -l
[12:55:43 AM] [FlutterDaemon] [Error] [  +26 ms] List of devices attached
[12:55:47 AM] [FlutterDaemon] [Info] ==> [{"id":"4","method":"daemon.getSupportedPlatforms","params":{"projectRoot":"E:\\works\\dp2\\rx"}}]
[12:55:47 AM] [FlutterDaemon] [Error] [+4015 ms] executing: C:\Users\namka\AppData\Local\Android\sdk\platform-tools\adb.exe devices -l
[12:55:47 AM] [FlutterDaemon] [Error] [  +26 ms] List of devices attached
[12:55:51 AM] [FlutterDaemon] [Error] [+4008 ms] executing: C:\Users\namka\AppData\Local\Android\sdk\platform-tools\adb.exe devices -l
[12:55:51 AM] [FlutterDaemon] [Error] [  +25 ms] List of devices attached
[12:55:52 AM] [FlutterDaemon] [Info] ==> [{"id":"5","method":"emulator.getEmulators"}]
[12:55:52 AM] [FlutterDaemon] [Info] ==> [{"id":"6","method":"emulator.getEmulators"}]
[12:55:55 AM] [FlutterDaemon] [Error] [+4007 ms] executing: C:\Users\namka\AppData\Local\Android\sdk\platform-tools\adb.exe devices -l
[12:55:55 AM] [FlutterDaemon] [Error] [  +26 ms] List of devices attached
[12:55:56 AM] [FlutterDaemon] [Info] Spawning E:\works\flutter\bin\flutter.bat with args ["config","--machine"] in E:\works\flutter with env undefined
[12:55:59 AM] [FlutterDaemon] [Error] [+4006 ms] executing: C:\Users\namka\AppData\Local\Android\sdk\platform-tools\adb.exe devices -l
[12:55:59 AM] [FlutterDaemon] [Error] [  +26 ms] List of devices attached
[12:56:00 AM] [FlutterDaemon] [Info] ==> [{"id":"7","method":"daemon.getSupportedPlatforms","params":{"projectRoot":"E:\\works\\dp2\\rx"}}]
[12:56:03 AM] [FlutterDaemon] [Error] [+4001 ms] executing: C:\Users\namka\AppData\Local\Android\sdk\platform-tools\adb.exe devices -l
[12:56:03 AM] [FlutterDaemon] [Error] [  +26 ms] List of devices attached
[12:56:07 AM] [FlutterDaemon] [Error] [+4010 ms] executing: C:\Users\namka\AppData\Local\Android\sdk\platform-tools\adb.exe devices -l
[12:56:07 AM] [FlutterDaemon] [Error] [  +25 ms] List of devices attached
[12:56:11 AM] [FlutterDaemon] [Error] [+4009 ms] executing: C:\Users\namka\AppData\Local\Android\sdk\platform-tools\adb.exe devices -l
[12:56:11 AM] [FlutterDaemon] [Error] [  +25 ms] List of devices attached
[12:56:15 AM] [FlutterDaemon] [Error] [+4015 ms] executing: C:\Users\namka\AppData\Local\Android\sdk\platform-tools\adb.exe devices -l
[12:56:15 AM] [FlutterDaemon] [Error] [  +25 ms] List of devices attached
[12:56:17 AM] [FlutterDaemon] [Info] ==> [{"id":"8","method":"daemon.getSupportedPlatforms","params":{"projectRoot":"E:\\works\\dp2\\rx"}}]
[12:56:19 AM] [FlutterDaemon] [Error] [+4010 ms] executing: C:\Users\namka\AppData\Local\Android\sdk\platform-tools\adb.exe devices -l
[12:56:19 AM] [FlutterDaemon] [Error] [  +26 ms] List of devices attached
[12:56:22 AM] [FlutterDaemon] [Info] ==> [{"id":"9","method":"emulator.getEmulators"}]
[12:56:23 AM] [FlutterDaemon] [Error] [+4015 ms] executing: C:\Users\namka\AppData\Local\Android\sdk\platform-tools\adb.exe devices -l
[12:56:23 AM] [FlutterDaemon] [Error] [  +26 ms] List of devices attached
[12:56:27 AM] [FlutterDaemon] [Error] [+4008 ms] executing: C:\Users\namka\AppData\Local\Android\sdk\platform-tools\adb.exe devices -l
[12:56:27 AM] [FlutterDaemon] [Error] [  +26 ms] List of devices attached
[12:56:31 AM] [FlutterDaemon] [Error] [+4013 ms] executing: C:\Users\namka\AppData\Local\Android\sdk\platform-tools\adb.exe devices -l
[12:56:32 AM] [FlutterDaemon] [Error] [  +28 ms] List of devices attached
[12:56:36 AM] [FlutterDaemon] [Error] [+4002 ms] executing: C:\Users\namka\AppData\Local\Android\sdk\platform-tools\adb.exe devices -l
[12:56:36 AM] [FlutterDaemon] [Error] [  +27 ms] List of devices attached
[12:56:40 AM] [FlutterDaemon] [Error] [+4001 ms] executing: C:\Users\namka\AppData\Local\Android\sdk\platform-tools\adb.exe devices -l
[12:56:40 AM] [FlutterDaemon] [Error] [  +26 ms] List of devices attached
[12:56:44 AM] [FlutterDaemon] [Error] [+4016 ms] executing: C:\Users\namka\AppData\Local\Android\sdk\platform-tools\adb.exe devices -l
[12:56:44 AM] [FlutterDaemon] [Error] [  +42 ms] List of devices attached
[12:56:48 AM] [FlutterDaemon] [Error] [+4012 ms] executing: C:\Users\namka\AppData\Local\Android\sdk\platform-tools\adb.exe devices -l
[12:56:48 AM] [FlutterDaemon] [Error] [  +26 ms] List of devices attached
[12:56:52 AM] [FlutterDaemon] [Error] [+4007 ms] executing: C:\Users\namka\AppData\Local\Android\sdk\platform-tools\adb.exe devices -l
[12:56:52 AM] [FlutterDaemon] [Error] [  +26 ms] List of devices attached
[12:56:56 AM] [FlutterDaemon] [Error] [+4004 ms] executing: C:\Users\namka\AppData\Local\Android\sdk\platform-tools\adb.exe devices -l
[12:56:56 AM] [FlutterDaemon] [Error] [  +26 ms] List of devices attached
[12:57:00 AM] [FlutterDaemon] [Error] [+4008 ms] executing: C:\Users\namka\AppData\Local\Android\sdk\platform-tools\adb.exe devices -l
[12:57:00 AM] [FlutterDaemon] [Error] [  +26 ms] List of devices attached

</details>

**Comment by DanTup** [comment]
@namkazt it looks like the daemon did respond to some of the requests from VS Code before it stopped. I think it's the same issue, but depending on how quickly the stdin stream closes, there might be some completed requests.

I'll try follow up on https://github.com/Dart-Code/Dart-Code/issues/4840 soon. We've managed to remove VS Code + Dart-Code and use a NodeJS script to reproduce it, and the next step is trying to reduce the Flutter daemon to something simpler (to try and eliminate bugs in the layers it has over the stdin stream). Unfortunately I haven't gotten back to this yet.

**Comment by huycozy** [comment]
Since it's unclear if this is related to Flutter side (tool) or not, (see Danny's comment https://github.com/Dart-Code/Dart-Code/issues/4840#issuecomment-1930605807) I will temporarily add the label `waiting for customer response` to keep this issue open. Once there is progress at https://github.com/Dart-Code/Dart-Code/issues/4840, please help update here. Thanks!

**Comment by namkazt** [comment]
thanks, it probadly cause from flutter_tool daemon stdin.

**Comment by namkazt** [comment]
yes, it is listed and deamon working fine in windows command or powershell.

![image](https://github.com/flutter/flutter/assets/6035916/89b5d331-d816-4d69-bc09-a0d80a889c41)

Daemon not working with android studio either. i tried.

**Comment by huycozy** [comment]
Thanks for checking. I checked this on my Windows machine but the issue is not reproducible there. Labeling the issue for other's input.

**Comment by namkazt** [comment]
> hi guys,
> 
> I have now this issue - would be glad if you can help me.
> 
> I switched from intel Mac to M3 Mac and I migrated the data from the old one to the new one. I reinstalled then VSCode, flutter sdk, Xcode(via appstore) and android studio to have Silicon installations. Now I am stuck with this "Starting device daemon..." and of course the simulators are not able to boot up.
> 
> Maybe you guys can help me?

@paulunder  you should follow this issue: https://github.com/Dart-Code/Dart-Code/issues/4840

there is few workarounds if you want a dev env immedately.

**Comment by DanTup** [comment]
> you should follow this issue: https://github.com/Dart-Code/Dart-Code/issues/4840

So far my understanding of that issue is that it's Windows-specific. @paulunder I would be interested in a copy of the [Flutter Device Daemon logfile](https://dartcode.org/docs/logging/#flutter-daemon) for your case in a new issue at https://github.com/Dart-Code/Dart-Code/issues/

Thanks!

**Comment by DanTup** [comment]
I have a repro for this issue now thanks to https://github.com/Dart-Code/Dart-Code/issues/4840#issuecomment-2120605265

It appears that having msys2 on `PATH` (possibly `git` specifically, though I haven't confirmed that yet) causes this issue. I haven't gotten to the bottom of _why_ yet, but I am interested to know whether others seeing this issue have msys2 on `PATH`. In particular, if you run the following in a PowerShell window, so you see a path that contains something like `msys64\usr\bin`?

```
echo $env:PATH
```

And also, what you see if you run:

```
where.exe git
```

Thanks!

**Comment by mainjzb** [comment]
same problem

**Comment by DanTup** [comment]
@mainjzb it's not clear to me what problem you're seeing - the daemon log you posted for VS Code insiders seems to show the daemon working (it sends devices and responds to requests for the supported platforms). 

The best current workaround for this issue is to not have Git from msys2 on `PATH`, but instead ensure you have a standard Windows version of Git installed (and first in your `PATH`).

I don't understand the reason for the failure but I opened https://github.com/dart-lang/sdk/issues/55792 in the Dart SDK with a repro so this issue is currently blocked on that.

**Comment by BinniZenobioCordovaLeandro** [comment]
Same issue here.
I am trying to debug on an external device, an Android Honor tablet from my MacBook M1
and I see the flutter is stuck in:

Starting device daemon...
![Screenshot 2024-12-08 at 9 36 38 PM](https://github.com/user-attachments/assets/c0bc9006-1016-4e4a-b769-2c7f1a32803b)

after building and installing the .apk on the tablet.
![Screenshot 2024-12-08 at 9 37 00 PM](https://github.com/user-attachments/assets/7ce39873-cc76-4e23-9622-80571ac3fa14)

both devices are in the same network, and curiously if I use an emulator the `debug` stars normally.

**Comment by paulunder** [comment]
Hey @BinniZenobioCordovaLeandro 

For me the only solution that worked - I had to delete the cache in my xcode devices - maybe you have to do something similar in android studio?

**Comment by DanTup** [comment]
@BinniZenobioCordovaLeandro sorry for the delay, I thought I had replied here.

The screenshot shown above that shows "Starting device daemon..." is not actually a problem. This Output panel just shows the output from the Flutter device daemon, and it seems that it prints this text, but never prints anything else _even if it has finished starting up_. So that is not itself a problem.

The second screenshot (assuming it is stuck on "Funning Gradle task") is a different issue and should be filed separately. If the same occurs when running from the terminal with `flutter run`, then it should be filed in the Flutter repo.

**Comment by DanTup** [comment]
I've opened https://github.com/flutter/flutter/pull/160444 to change the text printed when the daemon starts up. I think the current text is confusing and makes it seem like other issues are related.

When that lands, I will close this - because most of the discussion is about that text making it seem like the daemon hasn't started up. If you are seeing issues other than just this text label making it _seem_ like the daemon didn't start up, please file separate issues, with a clear description of that issue (ignoring this text).

Thanks!
